inference-engine/cmake/dependencies.cmake

```diff
@@ -188,91 +188,91 @@ if (THREADING STREQUAL "TBB" OR THREADING STREQUAL "TBB_AUTO")
 endif ()
 
 if (ENABLE_OPENCV)
-    reset_deps_cache(OpenCV_DIR)
-
-    set(OPENCV_VERSION "4.5.1")
-    set(OPENCV_BUILD "044")
-    set(OPENCV_BUILD_YOCTO "337")
-
-    if (AARCH64)
-        if(DEFINED ENV{THIRDPARTY_SERVER_PATH})
-            set(IE_PATH_TO_DEPS "$ENV{THIRDPARTY_SERVER_PATH}")
-        elseif(DEFINED THIRDPARTY_SERVER_PATH)
-            set(IE_PATH_TO_DEPS "${THIRDPARTY_SERVER_PATH}")
-        else()
-            message(WARNING "OpenCV is not found!")
-        endif()
-
-        if(DEFINED IE_PATH_TO_DEPS)
-            set(OPENCV_SUFFIX "yocto_kmb")
-            set(OPENCV_BUILD "${OPENCV_BUILD_YOCTO}")
-
-            RESOLVE_DEPENDENCY(OPENCV
-                    ARCHIVE_LIN "opencv/opencv_${OPENCV_VERSION}-${OPENCV_BUILD}_${OPENCV_SUFFIX}.txz"
-                    TARGET_PATH "${TEMP}/opencv_${OPENCV_VERSION}_${OPENCV_SUFFIX}/opencv"
-                    ENVIRONMENT "OpenCV_DIR"
-                    VERSION_REGEX ".*_([0-9]+.[0-9]+.[0-9]+).*"
-                    SHA256 "b5239e0e50b9009f95a29cb11f0840ec085fa07f6c4d3349adf090f1e51b0787")
+    if(NOT DEFINED OpenCV_DIR AND NOT DEFINED ENV{OpenCV_DIR})
```

Comment:
I do not see `find_package(OpenCV)` statement in the `else` branch? Is it called somewhere else?

> OpenCV_DIR

Not sure if we should stick to this variable only.
Please note that there are several other options how to make `find_package(OpenCV)` besides of `OpenCV_DIR`:
- modern CMake way: `OpenCV_ROOT`
- automatic `$PATH/../cmake` locations
- common `CMAKE_PREFIX_PATH` (useful for cross-compilation)
- other methods: [find_package](https://cmake.org/cmake/help/latest/command/find_package.html)

---

> reset_deps_cache(OpenCV_DIR)

What is happen on the second run of CMake: `cmake .` ?

Reply:
Why `reset_deps_cache(OpenCV_DIR)` mechanism doesn't work, it was developed exactly for this reason (allow to use external dependencies or update internal).

`reset_deps_cache` checks this provided variable. If it points inside `TEMP` directory (where OpenVINO CMake puts all internal artifacts), it resets the variable. Otherwise, it leaves it untouched.

`RESOLVE_DEPENDENCY` also checks if `OpenCV_DIR` variable. If it is set, it skips package download.

So, it `OpenCV_DIR` was manually set by user to custom location, `reset_deps_cache` will no touch it, `RESOLVE_DEPENDENCY` will do nothing and `find_package` will return this custom OpenCV project.

If it doesn't work, this means that there is some bug in the mechnism.

Reply:
@vinograd47 If I install custom OpenCV into subdirectory in TEMP directory and define OpenCV_DIR to this subdirectory the variable will be reset?

Reply:
Yes. But, IMHO, this scenario is a bit strange. OpenVINO TEMP directory is internal for OpenVINO and is managed by it. Why to put something to it from outside? Even its name points that this directory might not be persistent.

Reply:
@vinograd47 I used this scenario for cross compilation. There are some problems in our environment with order of libraries that cmake find as dependency and custom builds libraries like OpenCV could not be found if they are placed out of openvino source tree.

I agree that it is not correct way to place them into temp directory. CMAKE_STAGING_PREFIX should be used instead.

So I am going to close the PR.